osu.Game/Skinning/LegacySkin.cs

```diff
@@ -38,6 +38,11 @@ public class LegacySkin : Skin
 
         protected virtual bool AllowManiaSkin => hasKeyTexture.Value;
 
+        /// <summary>
+        /// Whether this skin will fall back to default HUD components if it has no fonts for use.
+        /// </summary>
+        protected virtual bool AllowDefaultHUDComponentsFallback => true;
```

Comment:
I don't think this pattern of virtual `bool`s is working out in this class. Worried it'll spiral out of control with another usage or two.

Maybe worth to at least try removing this and overriding `GetDrawableComponent()` in `LegacyBeatmapSkin` directly to get the desired behaviour? Not necessarily saying it's the way to go yet, though.

Reply:
I was initially going the `GetDrawableComponent` override way but wasn't feeling 100% about it given it completely overrides the method just to go to that one component lookup case and return null on it, not to mention how to make sense of it.

As for usages, `AllowDefaultHUDComponentsFallback` would probably not be used anywhere else at all, and will vanish away with falling back to default legacy skin instead.

Reply:
Actually, if it's gonna be removed soon either way, let's do it the override way and clear up the confusion with that property.